### ISSUE TITLE
[Bug] Different results between inference on test videos and using long_video_demo.py

### ISSUE THREAD

**Issue by zzz77575** [open-mmlab/mmaction2#2911]
### Branch

main branch (1.x version, such as `v1.0.0`, or `dev-1.x` branch)

### Prerequisite

- [x] I have searched [Issues](https://github.com/open-mmlab/mmaction2/issues) and [Discussions](https://github.com/open-mmlab/mmaction2/discussions) but cannot get the expected help.
- [x] I have read the [documentation](https://mmaction2.readthedocs.io/en/latest/) but cannot get the expected help.
- [x] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmaction2).

### Environment

Python 3.8.18
Package                Version      
---------------------- ------------
addict                 2.4.0
aliyun-python-sdk-core 2.16.0
aliyun-python-sdk-kms  2.16.5
asttokens              3.0.0
backcall               0.2.0
certifi                2025.7.14
cffi                   1.17.1
charset-normalizer     3.4.2
chumpy                 0.70
click                  8.1.8
colorama               0.4.6
comm                   0.2.2
contourpy              1.1.1
crcmod                 1.7
cryptography           45.0.5
cycler                 0.12.1
Cython                 3.1.2
debugpy                1.6.7
decorator              4.4.2
decord                 0.6.0
einops                 0.8.1
et_xmlfile             2.0.0
executing              2.1.0
ffmpeg                 1.4
ffmpeg-python          0.2.0
filelock               3.14.0
fonttools              4.57.0
future                 1.0.0
idna                   3.10
imageio                2.35.1
imageio-ffmpeg         0.5.1
importlib_metadata     8.5.0
importlib_resources    6.4.5
ipykernel              6.29.5
ipython                8.12.2
jedi                   0.19.1
jmespath               0.10.0
joblib                 1.4.2
json-tricks            3.17.3
jupyter_client         8.6.3
jupyter_core           5.8.1
kiwisolver             1.4.7
Markdown               3.7
markdown-it-py         3.0.0
matplotlib             3.7.5
matplotlib-inline      0.1.7
mdurl                  0.1.2
mmaction2              1.2.0      
mmcv                   2.0.1
mmdet                  3.3.0
mmengine               0.10.7
mmpose                 1.3.2
model-index            0.1.11
moviepy                1.0.3
munkres                1.1.4
nest_asyncio           1.6.0
numpy                  1.24.4
opencv-contrib-python  4.12.0.88
opencv-python          4.12.0.88
opendatalab            0.0.10
openmim                0.3.9
openpyxl               3.1.5
openxlab               0.1.2
ordered-set            4.1.0
oss2                   2.17.0
packaging              24.2
pandas                 2.0.3
parso                  0.8.4
pexpect                4.9.0
pickleshare            0.7.5
pillow                 10.4.0
pip                    24.2
platformdirs           4.3.6
proglog                0.1.12
prompt_toolkit         3.0.48
psutil                 6.0.0
ptyprocess             0.7.0
pure_eval              0.2.3
pycocotools            2.0.7
pycparser              2.22
pycryptodome           3.23.0
Pygments               2.18.0
pyparsing              3.1.4
python-dateutil        2.9.0
pytz                   2023.4
PyYAML                 6.0.2
pyzmq                  25.1.2
requests               2.28.2
rich                   13.4.2
scikit-learn           1.3.2
scipy                  1.10.1
seaborn                0.13.2
setuptools             60.2.0
shapely                2.0.7
six                    1.16.0
stack-data             0.6.2
tabulate               0.9.0
termcolor              2.4.0
terminaltables         3.1.10
threadpoolctl          3.5.0
tomli                  2.2.1
torch                  1.13.0+cu117
torchaudio             0.13.0+cu117
torchvision            0.14.0+cu117
tornado                6.4.1
tqdm                   4.65.2
traitlets              5.14.3
typing_extensions      4.12.2
tzdata                 2025.2
urllib3                1.26.20
wcwidth                0.2.13
wheel                  0.44.0
xtcocotools            1.14.3
yapf                   0.43.0
zipp                   3.21.0

### Describe the bug

I attempted to use checkpoints trained for action recognition to perform inference on long videos. When comparing the results obtained by directly testing on short clips using `tools/test.py` and concatenating the results, versus performing sliding window inference on the entire video using `demo/long_video_demo.py`, there was a significant difference in the results. The test results showed high accuracy, but the demo script's results were poor. I am unsure whether the issue lies with the inference method or the sliding window implementation. I tried aligning the sliding window methods of both approaches and experimented with different parameters in the `test_pipeline` function in `long_video_demo.py`, but saw no improvement. Additionally, the `sample_length` parameter in `long_video_demo.py` was not multiplied by `frame_interval`, and even after fixing this issue, the results did not improve.
